### DISCUSSION TITLE
generate_network took forever

### DISCUSSION THREAD

**Discussion by zhouzhendiao** [tuonglab/dandelion#298] [Q&A] [answered]
After preprocessing,  codes below were excuted:
```
ddl.tl.find_clones(vdj)

Finding clones based on B cell VDJ chains : 100%|██████████| 500/500 [00:01<00:00, 422.03it/s]
Refining clone assignment based on VJ chain pairing : 100%|██████████| 42036/42036 [00:00<00:00, 189946.50it/s]
```
```
ddl.tl.generate_network(vdj)

Setting up data: 96034it [00:43, 2218.97it/s]
Calculating distances... : 100%|██████████| 16436/16436 [00:40<00:00, 401.91it/s] 
```

"generate_network" seems to runing too long, and nearly no cpus are using(<5%, single core).

**Comment by zktuong** [comment, marked as answer]
Hi, the slowdown is most likely link to the layout generation step - i don’t have a good solution unfortunately. You can try downsampling your cell numbers to something more manageable or try the graph-tool sfdp layout option, although that has its own issues.

**Reply by zhouzhendiao** [reply]
OK, I got it. Thanks!

**Reply by zktuong** [reply]
actually, i found the issue. the latest version on github has solved this problem

**Reply by zhouzhendiao** [reply]
OK, thanks!
